realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -1809,6 +1816,139 @@ public void run() {
         }.start();
     }
 
+    // Test case for https://github.com/realm/realm-java/issues/2417
```

Comment:
Including the issue is great, but please also include a description of what it tests, so one doesn't have to lookup the issue. (same for the below issue references)